packages/react-native-client/src/hooks/useReconnection.ts

```diff
@@ -10,21 +11,9 @@ export type ReconnectionStatus = 'idle' | 'reconnecting' | 'error';
  * @category Connection
  */
 export function useReconnection() {
-  const [reconnectionStatus, setReconnectionStatus] =
-    useState<ReconnectionStatus>('idle');
-
-  const setStatusIdle = useCallback(() => setReconnectionStatus('idle'), []);
-  const setStatusReconnecting = useCallback(
-    () => setReconnectionStatus('reconnecting'),
-    [],
-  );
-  const setStatusError = useCallback(() => setReconnectionStatus('error'), []);
-
-  useFishjamEvent(ReceivableEvents.Reconnected, setStatusIdle);
-  useFishjamEvent(ReceivableEvents.ReconnectionStarted, setStatusReconnecting);
-  useFishjamEvent(
-    ReceivableEvents.ReconnectionRetriesLimitReached,
-    setStatusError,
+  const reconnectionStatus = useFishjamEventState<ReconnectionStatus>(
+    ReceivableEvents.ReconnectionStatusChanged,
+    RNFishjamClientModule.reconnectionStatus,
```

Comment:
I wonder if we should consider some internal additional guards to protect against receiving incorrect values from native part. LIke some debug-time checking if `RNFishjamClientModule.reconnectionStatus` will have only expected values 🤔 (but this is not for this PR)

cc @MiloszFilimowski

Reply:
Agreed. This also came up during `usePeerStatus` although we decided that it's not an issue at the moment